### PR TITLE
feat: arm64 arch support for snap

### PR DESCRIPTION
**Pull request by yanksyoon**
This change adds arm64 support for aproxy.

### CODE REVIEW

**Comment by weiiwang01** [comment]
It seems like snapcraft skipped the arm64 build on amd64 machines: https://github.com/canonical/aproxy/actions/runs/7041282923/job/19163572832?pr=7#step:4:74

Golang should support cross-compiling, but I think snap's go plugin doesn't support that, let me think if there's any solution to make this work.

**Comment by yanksyoon** [comment]
Could it have been skipped because the runner is amd?

**Comment by weiiwang01** [comment]
> Could it have been skipped because the runner is amd?

Yes, I think this is the reason.

**Comment by yanksyoon** [comment]
@weiiwang01 I'll parametrize the tests and have one running on Github's own arm64 machines :D

**Comment by yanksyoon** [comment]
@weiiwang01 oh nvm gh doesn't have arm machines

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -23,11 +28,18 @@ apps:
...
     source: .
     build-snaps:
       - go
+    override-build: |
+      snapcraftctl build
+      GOARCH=$SNAPCRAFT_TARGET_ARCH
+      CGO_ENABLED=0
+      go mod download
+      go build -ldflags="-w -s"
+      cp aproxy ${SNAPCRAFT_PART_INSTALL}/bin
```
Comment (weiiwang01) [inline snap/snapcraft.yaml:42]:
```suggestion
      mkdir ${SNAPCRAFT_PART_INSTALL}/bin
      cp aproxy ${SNAPCRAFT_PART_INSTALL}/bin/aproxy
```
Sorry, this should fix the issue.

Comment (yanksyoon) [reply snap/snapcraft.yaml:43]: Thanks for the prompt fix! Updated!

**Review by weiiwang01 — APPROVED** [review]

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -23,11 +28,19 @@ apps:
...
-    plugin: go
+  aproxy:
+    plugin: nil
     source: .
     build-snaps:
       - go
+    override-build: |
+      snapcraftctl build
+      GOARCH=$SNAPCRAFT_TARGET_ARCH
+      CGO_ENABLED=0
```
Comment (weiiwang01) [inline snap/snapcraft.yaml:39]:
```suggestion
      export GOARCH=$SNAPCRAFT_TARGET_ARCH
      export CGO_ENABLED=0
```
Oops, sorry about this.

**Pull request merged**